pkg/ingester/user_tsdb.go

```diff
@@ -624,7 +624,9 @@ func (u *userTSDB) computeOwnedSeries() int {
 			if u.ownedTokenRanges.IncludesKey(sh) {
 				count++
 			} else {
-				u.activeSeries.Delete(refs[i])
+				idx, _ := u.Head().Index()
```

Comment:
Also here, please retrieve the index only once.

Reply:
addressed in commit https://github.com/grafana/mimir/commit/f4ebc07871ef79a0ed7ee31c48032a9fd42fe757#diff-e1032332627c413a3010c66b54b22b6e9835cf152fa339e40cf0b11204f7241fR1431